core/src/main/scala/org/apache/spark/deploy/SparkSubmit.scala

```diff
@@ -1005,17 +1005,20 @@ private[spark] class SparkSubmit extends Logging {
         e
     }
 
+    var exitCode = 0
+
     try {
       app.start(childArgs.toArray, sparkConf)
     } catch {
       case t: Throwable =>
+        exitCode = -1
         throw findCause(t)
     } finally {
       if (args.master.startsWith("k8s") && !isShell(args.primaryResource) &&
           !isSqlShell(args.mainClass) && !isThriftServer(args.mainClass) &&
           !isConnectServer(args.mainClass)) {
         try {
-          SparkContext.getActive.foreach(_.stop())
+          SparkContext.getActive.foreach(_.stop(exitCode))
```

Comment:
I don't think this is related to YARN AM because this is guarded by `if (args.master.startsWith("k8s")`. Is this K8s patch instead of YARN AM?

Reply:
Code in spark-3.1.2 is 
```
      if (!isShell(args.primaryResource) && !isSqlShell(args.mainClass) &&
        !isThriftServer(args.mainClass)) {
```

Seems https://github.com/apache/spark/pull/33403 change the behavior

Reply:
I don't know too much about K8S scheduler, but for yarn client mode we also need to keep a same exit code.